sqldelight/src/main/java/org/ergoplatform/persistance/SqlDelightAppDb.kt

```diff
@@ -1,8 +1,19 @@
 package org.ergoplatform.persistance
 
-class SqlDelightAppDb(private val appDatabase: AppDatabase): IAppDatabase {
+class SqlDelightAppDb(
+    val appDatabase: AppDatabase,
+    /**
+     * if set to false, no transactions will be used for SQlite operations.
+     * Use on iOS as its SQlite version cannot handle transactions
```

Comment:
Where this statement "its SQlite version cannot handle transactions" comes from? Sounds unrealistic to me.

Reply:
It can handle transactions, but only up to a certain amount of changes. sql statement execution is freezing since the addition of transaction list when an address with transactions is deleted with transactions enabled on iOS. The iOS SQlite support is very basic as Apple wants to push its own data storage solution.

Reply:
The "freezing" can be solved by doing transaction in background thread (not UI thread). Transactions ensure atomicity of changes which ensure consistency of the data. 
Data inconsistency on user device is really hard to diagnose and in many cases you will not even know about it. Users will just do data refresh and be annoyed by the app doing something weird.

Reply:
I haven't explained well enough. Of course the db access is made on an IO thread, but that is probably the root cause. The UI is still responding and the waiting spinner animating, but the SQL execute statement never comes back. 
Reading up more, I assume that this is caused by coroutines' IO dispatcher using different threads to access the database which is not allowed within SQlite transactions.

I also consider using transactions as better, I don't think they are very important in a mobile app as we have a single user connection here. The transactions are mostly a convinience layer to handle rollbacks for thrown exceptions in batch processing. In our case, this is mostly for deletions as all other data updates are done on a single row and table anyway.

I will check if the problem is really caused by the IO dispatcher by using the Main dispatcher for a test. If this helps, using an own dispatcher might solve the problem.

Reply:
Update: nested coroutines are the cause. I could confirm this on the not-finished desktop implementation, and luckily in contrast to iOS, the desktop SQlight version throws an `SQLiteException: [SQLITE_BUSY]  The database file is locked (database is locked)` instead of going into a freeze.

Reply:
I was able to work around the issue